src/rust/engine/remote_provider/remote_provider_reapi/src/byte_store_tests.rs

```diff
@@ -436,6 +436,23 @@ async fn list_missing_digests_none_missing() {
     )
 }
 
+#[tokio::test]
+async fn list_missing_digests_more_than_4mb() {
+    let testdata = TestData::roland();
+    let _ = WorkunitStore::setup_for_tests();
+    let cas = StubCAS::builder().file(&testdata).build();
+
+    let provider = new_provider(&cas).await;
+
+    let test_data = (0..100_000).map(|_| testdata.digest()).collect::<Vec<_>>();
+    assert_eq!(
+        provider
+            .list_missing_digests(&mut test_data.into_iter())
+            .await,
+        Ok(HashSet::new())
```

Comment:
Thanks for the test. I think there's a few plausible mistakes that we could introduce in future that won't be caught:

1. only sending each unique digest once, thus invalidating this test (i.e. if we do the `HashSet` idea, the `list_missing_digests` will be working with just one digest and thus only send one request)
2. skipping the second+ request (either not sending them at all, or not including the results in the final aggregation)
3. Not sure about this one: if we break this in another way and start sending >4MiB requests, does the `StubCAS` reject the requests (i.e. does this test fail without the fix)?


I think we can address each of them:

1. check `assert_eq!(cas.request_count(RequestType::CASFindMissingBlobs), 2)` so that the test starts failing if that core assumption changes
2. add two digests that is missing, within each "chunk" (e.g. one at the start and one at the end)
3. add request size enforcement to `StubCAS` if not already there (this might cause other tests to fail, so we might want it to be optional until we resolve them)